zkvm/src/predicate.rs

```diff
@@ -121,6 +121,15 @@ impl Predicate {
     }
 }
 
+impl From<Scalar> for Predicate {
```

Comment:
Contrary to other conversions, this seems to me a little bit off. Scalar is not a _representation_ of a Predicate. It's an input of a specific subkind of a predicate. At the call site it may not be super clear why and where some scalar is being converted to with `x.into()`. 

How about something like `Predicate::from_secret_key(Scalar)` to make it more clear what's the meaning of the scalar and where it's being converted?

Reply:
I like that!